File: apps/frontend/src/types/opencart.ts
export interface OpenCartMatchSettings {
  use_openrouter: boolean;
  model: string;
  fuzzy_threshold: number;
  low_confidence_threshold: number;
  ignore_service_words: boolean;
}

export interface OpenCartGenerateRequest {
  products_text: string;
  files_text: string;
  image_prefix: string;
  settings: OpenCartMatchSettings;
  openrouter_api_key?: string;
}

export interface OpenCartProduct {
  product_id: number;
  sku: string;
  line_number: number;
}

export interface OpenCartParseError {
  line_number: number;
  line: string;
  message: string;
}

export interface OpenCartImageMatch {
  product_id: number;
  sku: string;
  filename: string;
  image_path: string;
  method: string;
  confidence: number;
  reason: string;
}

export interface OpenCartMatchConflict {
  product_id: number | null;
  sku: string | null;
  filename: string | null;
  message: string;
}

export interface OpenCartGenerateResponse {
  history_id: string;
  matches: OpenCartImageMatch[];
  unmatched_products: OpenCartProduct[];
  unused_files: string[];
  parse_errors: OpenCartParseError[];
  conflicts: OpenCartMatchConflict[];
  low_confidence_matches: OpenCartImageMatch[];
  sql: string;
}

export interface OpenCartHistorySummary {
  id: string;
  created_at: string;
  total_products: number;
  total_files: number;
  matched_count: number;
  unmatched_count: number;
  unused_file_count: number;
  used_openrouter: boolean;
  model: string | null;
}

export interface OpenCartHistoryList {
  items: OpenCartHistorySummary[];
  total: number;
  page: number;
  page_size: number;
}

export interface OpenCartHistoryDetail extends OpenCartHistorySummary {
  products_text: string;
  files_text: string;
  image_prefix: string;
  settings: OpenCartMatchSettings;
  result: OpenCartGenerateResponse;
  sql: string;
}

File: apps/backend/src/application/dto/opencart.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, Field


class OpenCartMatchSettingsDTO(BaseModel):
    use_openrouter: bool = False
    model: str = "openai/gpt-4.1-nano"
    fuzzy_threshold: float = Field(default=0.78, ge=0.0, le=1.0)
    low_confidence_threshold: float = Field(default=0.86, ge=0.0, le=1.0)
    ignore_service_words: bool = True


class OpenCartGenerateRequestDTO(BaseModel):
    products_text: str = Field(min_length=1)
    files_text: str = Field(min_length=1)
    image_prefix: str = ""
    settings: OpenCartMatchSettingsDTO = Field(default_factory=OpenCartMatchSettingsDTO)
    openrouter_api_key: str | None = None


class OpenCartProductDTO(BaseModel):
    product_id: int
    sku: str
    line_number: int


class OpenCartParseErrorDTO(BaseModel):
    line_number: int
    line: str
    message: str


class OpenCartImageMatchDTO(BaseModel):
    product_id: int
    sku: str
    filename: str
    image_path: str
    method: str
    confidence: float
    reason: str


class OpenCartMatchConflictDTO(BaseModel):
    product_id: int | None = None
    sku: str | None = None
    filename: str | None = None
    message: str


class OpenCartGenerateResponseDTO(BaseModel):
    history_id: UUID
    matches: list[OpenCartImageMatchDTO]
    unmatched_products: list[OpenCartProductDTO]
    unused_files: list[str]
    parse_errors: list[OpenCartParseErrorDTO]
    conflicts: list[OpenCartMatchConflictDTO]
    low_confidence_matches: list[OpenCartImageMatchDTO]
    sql: str


class OpenCartHistorySummaryDTO(BaseModel):
    id: UUID
    created_at: datetime
    total_products: int
    total_files: int
    matched_count: int
    unmatched_count: int
    unused_file_count: int
    used_openrouter: bool
    model: str | None = None


class OpenCartHistoryListDTO(BaseModel):
    items: list[OpenCartHistorySummaryDTO]
    total: int
    page: int
    page_size: int


class OpenCartHistoryDetailDTO(OpenCartHistorySummaryDTO):
    products_text: str
    files_text: str
    image_prefix: str
    settings: dict[str, Any]
    result: dict[str, Any]
    sql: str
